### ISSUE TITLE
Slow performance with Scaling

### ISSUE THREAD

**Issue by hub2000** [jwa107/Photoshop-Export-Layers-to-Files-Fast#121]
Hello,
we have 200 layers, when exporting with option "Scale 50%", then for each layer the image is resized by 50%.

The problem is, that for every single layer the whole psd image is resized with 200 layers (200 times resize of 200 layers instead of 200 times resize of 1 layer) , so it is very slow at the end.

**Comment by jwa107** [comment]
You're right it should probably resize the document and then export, that would make it more efficient!

**Comment by warmwhisky** [comment]
Is this related to exporting with many layers but using trim to size each image by the layer mask? I currently have over 400 layers, but it is only rendering the 8 I have visible. Each layer take approx. 4 minutes.

**Comment by hsw107** [comment]
Yes this would probably be the same thing, where it's resizing all layers at once over and over, which would be very slow for 400 layers.

**Comment by frostbitten** [comment]
> You're right it should probably resize the document and then export, that would make it more efficient!
> 
> For now I would maybe resize the document manually and then export without scaling.

This is what I do now but unfortunately this process is quite cumbersome:

1. One must first save the document anew to prevent overwriting the original artwork with an accidental save (This step can't be ignored).
2. Then resize.
3. Do the export process.
4. Close and open the correct document to edit again.
5. Did you make any last-minute edits, renames or reorganizations between steps 1-3? Manually merge or re-do that work in the original document.

It may seem like a simple process but it's quite a detriment to the workflow, requiring tedious work with pitfalls. I run into step 5 a lot. Not with every export but quite often there's at least one layer that needs renaming. Or sometimes worse I get distracted integrating the newly exported graphics and come back to edit without having done step 4, creating lots of work to redo.
